lib/matplotlib/axes/_axes.py

```diff
@@ -3048,6 +3065,208 @@ def broken_barh(self, xranges, yrange, **kwargs):
+                    labels=None, orientation="vertical", colors=None,
+                    **kwargs):
+        """
+        Make a grouped bar plot.
+
+        .. note::
+            This function is new in v3.10, and the API is still provisional.
+            We may still fine-tune some aspects based on user-feedback.
+
+        This is a convenience function to plot bar charts for multiple datasets
+        into one Axes. In particular, it simplifies positioning of the bars
+        compared to individual `~.Axes.bar` plots.
+
+        Parameters
+        ----------
+        x : array-like or list of str
+            The center positions of the bar groups. If these are numeric values,
+            they have to be equidistant. As with `~.Axes.bar`, you can provide
+            categorical labels, which will be used at integer numeric positions
+            ``range(x)``.
+
+        heights : list of array-like or dict of array-like or 2D array
+            The heights for all x and groups. One of:
+
+            - list of array-like: A list of datasets, each dataset must have
+              ``len(x)`` elements.
+
+              .. code-block:: none
+
+                  x = ["a", "b"]
+
+                  #            x[0]   x[1]
+                  dataset_0 = [ds0_a, ds0_b]
+                  dataset_1 = [ds1_a, ds1_b]
+                  dataset_2 = [ds2_a, ds2_b]
+
+                  heights = [dataset_0, dataset_1, dataset_2]
+
+              Example call::
+
+                  grouped_bar(x, [dataset_0, dataset_1, dataset_2])
+
+            - dict of array-like: A mapping names to datasets. Each dataset
+              (dict value) must have ``len(x)`` elements.
+
+              This is similar to passing a list of array-like, with the addition that
+              each dataset gets a name.
+
+              Example call::
+
+                grouped_bar(x, {'ds0': dataset_0, 'ds1': dataset_1, 'ds2': dataset_2]})
+
+              The names are used as *labels*, i.e. the following two calls are
+              equivalent::
+
+                data_dict = {'ds0': dataset_0, 'ds1': dataset_1, 'ds2': dataset_2]}
+                grouped_bar(x, data_dict)
+                grouped_bar(x, data_dict.values(), labels=data_dict.keys())
+
+              When using a dict-like input, you must not pass *labels* explicitly.
+
+            - a 2D array: The columns are the different datasets.
+
+              .. code-block:: none
+
+                          dataset_0 dataset_1 dataset_2
+                 x[0]="a"   ds0_a     ds1_a     ds2_a
+                 x[1]="b"   ds0_b     ds1_b     ds2_b
+
+              .. code-block::
+
+                  x = ["a", "b"]
+                  dataset_labels = ["dataset_0", "dataset_1", "dataset_2"]
+                  array = np.random.random((2, 3))
+
+              Note that this is consistent with pandas. These two calls produce
+              the same bar plot structure::
```

Comment:
```suggestion
              so if you created your array from a list of datasets, you will
              need to transpose it.  This is consistent with pandas. These two
              calls produce the same bar plot structure::
```
Since I am unfamiliar with pandas and am not used to thinking in terms of rows and columns I was unsure what to do here.  I appreciate it is already spelled out in the example but I, um, did not read the example before trying it out.

Reply:
This should be addressed through a rewriting below.